src/librustc_passes/ast_validation.rs

```diff
@@ -68,6 +68,26 @@ impl<'a> AstValidator<'a> {
             }
         }
     }
+
+    fn check_trait_fn_not_const(&self, span: Span, constness: Constness) {
+        match constness {
+            Constness::Const => {
+                // FIXME: This is just a wild guess, since the only accepted const fn
+                //        syntax is `const unsafe fn`, thus we attempt to span only the
+                //        first 5 characters on the function declaration's span.
+                //        Note that using `unsafe const fn` would throw a syntax error.
+                let span = Span {
+                    lo: span.lo,
+                    hi: BytePos(span.lo.0 + 5),
```

Comment:
Hmm, I am not very comfortable with this kind of span manipulation. :( It seems like it could easily cause problems when method declarations are created by macros and the like. If we want to track the span of the `Constness`, then I think we should add that into the AST (and/or HIR, as needed).